File: buoi_1/app/src/main/java/com/example/android23/MainActivity.java
package com.example.android23;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;


public class MainActivity<packageContext> extends AppCompatActivity {
    // khi tạo biến trong activity thì chúng ta để private
    private Button btnClickMe;
    private TextView tvName;
    int position = 0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        // Sau khi tạo biến ánh xạ biến đến button ở giao diện xml
        btnClickMe = findViewById(R.id.btnClickMe);
        tvName = findViewById(R.id.tvName);
        //1 Button - nút bấm, sự kiện bấm
        btnClickMe.setOnClickListener(view -> {
//                 khi bấm button, muốn set tên mình vào text view
//                 khi bấm button, các sự kiện mà mình muuosn sẽ được xử lý ở đây
//                 sử dụng settext để set dữ liệu vfao text view
//                 tvName.setText("a");


            //Khi ấn button tăng lên 1
//            position++;
//            // khi các biến float,.. + với 1 string-> kết qủa sẽ ra 1 string
//            tvName.setText("Đây là lần bấm thứ:" + position);
            Intent intent = new Intent(this,CodeFresher.class);
            startActivity(intent);
        });
    }
}
// sự khác nhau giữa int và integer là gì?
//kiếu số nguyên không có trường hợp NULL
//integer có trường hợp null( rất dễ xảy ra crash khi dùng)
// code ko đẹp thì dùng (Ctrl+Alt+L) -> format code(Ctrl+Alt+L)

File: btvn_buoi5/app/src/main/java/com/example/myapplication/MainActivity.kt
package com.example.myapplication

import android.app.DatePickerDialog
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.DatePicker
import android.widget.EditText
import android.widget.TextView
import androidx.core.widget.addTextChangedListener
import java.util.*

class MainActivity : AppCompatActivity(), DatePickerDialog.OnDateSetListener {

    private var personName: EditText? = null
    private var dateTime: EditText? = null
    private var sdt: EditText? = null
    private var gmail: EditText? = null
    private var cmt: EditText? = null
    private var v: TextView? = null
    private var personname: String = ""
    private var date: String = ""
    private var sdtt: String = ""
    private var email: String = ""
    private var cmtt: String = ""
    var day = 0
    var month = 0
    var year = 0
    var saveday = 0
    var savemoth = 0
    var saveyear = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        personName = findViewById(R.id.personName)
        dateTime = findViewById(R.id.dateTime)
        sdt = findViewById(R.id.sdt)
        gmail = findViewById(R.id.gmail)
        cmt = findViewById(R.id.cmt)
        v = findViewById(R.id.v)

        v?.isEnabled = false
        personName?.addTextChangedListener {
            personname = it.toString().trim()
            v?.isEnabled = personname.isNotEmpty() && date.isNotEmpty() && sdtt.isNotEmpty() && email.isNotEmpty() && cmtt.isNotEmpty()
        }

        dateTime?.addTextChangedListener {
            date = it.toString().trim()
            v?.isEnabled = personname.isNotEmpty() && date.isNotEmpty() && sdtt.isNotEmpty() && email.isNotEmpty() && cmtt.isNotEmpty()
        }
        sdt?.addTextChangedListener {
            sdtt = it.toString().trim()
            if (it?.length!! > 10) {
                sdt?.error = "Số điện thoại không hợp lệ"
            }
            v?.isEnabled = personname.isNotEmpty() && date.isNotEmpty() && sdtt.isNotEmpty() && email.isNotEmpty() && cmtt.isNotEmpty()
        }
        gmail?.addTextChangedListener {
            email = it.toString().trim()
            if (android.util.Patterns.EMAIL_ADDRESS.matcher(email).matches()) {

            } else {
                gmail?.error = "Email không hợp lệ"
            }
            v?.isEnabled = personname.isNotEmpty() && date.isNotEmpty() && sdtt.isNotEmpty() && email.isNotEmpty() && cmtt.isNotEmpty()
        }
        cmt?.addTextChangedListener {
            cmtt = it.toString().trim()
            if (it?.length != 12) {
                cmt?.error = "Chứng minh thư không hợp lệ"
            }
            v?.isEnabled = personname.isNotEmpty() && date.isNotEmpty() && sdtt.isNotEmpty() && email.isNotEmpty() && cmtt.isNotEmpty()
        }

        pickDate()
    }

    private fun getDateCalender() {
        val c = Calendar.getInstance()
        year = c.get(Calendar.YEAR)
        month = c.get(Calendar.MONTH)
        day = c.get(Calendar.DAY_OF_MONTH)
    }

    private fun pickDate() {
        dateTime?.setOnClickListener {
            getDateCalender()
            val dpd= DatePickerDialog(this,DatePickerDialog.OnDateSetListener{view, myear, mmonth, mday ->
                dateTime?.setText("$mday/${mmonth +1}/$myear")
            },year, month, day)
            dpd.show()
        }
    }
//    private fun pick() {
//        dateTime?.setOnClickListener {
//            getDateCalender()
//            val dpd= DatePickerDialog(this,DatePickerDialog.OnDateSetListener{view, myear, mmonth, mday ->
//                dateTime?.setText(""+ mday +"/"+ mmonth +"/" +myear)
//            },year, month, day)
//            dpd.show()
//        }
//    }
    override fun onDateSet(view: DatePicker?, year: Int, monthOfYear: Int, dayOfMonth: Int) {
        saveday = dayOfMonth
        savemoth = monthOfYear
        saveyear = year
        getDateCalender()
    }
}









